### PR TITLE
feat: [CDS-54465]: Reuse or create a new Perpetual Task for new Deployment -> code fixes and Unit Test

### PR DESCRIPTION
**Pull request by namantalaycha10**
## Describe your changes
Changes include some code fixes and Unit Test
Changes are backward compatible and are behind the FF
These feature is in progress so FF will only be enable when complete changes will be in Prod 

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/47465)

### CODE REVIEW

File: 126-instance/src/main/java/io/harness/service/instancesyncperpetualtask/instancesyncperpetualtaskhandler/k8s/K8SInstanceSyncPerpetualTaskHandler.java [hunk 1]
```diff
@@ -181,8 +181,8 @@ private K8sInstanceSyncPerpetualTaskParamsV2 createK8sInstanceSyncPerpetualTaskV
                             .build();
     return K8sInstanceSyncPerpetualTaskParamsV2.newBuilder()
         .setAccountId(infrastructureMappingDTO.getAccountIdentifier())
-        .setOrgId(connectorInfoDTO.getOrgIdentifier())
-        .setProjectId(connectorInfoDTO.getProjectIdentifier())
+        .setOrgId(connectorInfoDTO.getOrgIdentifier() == null ? "" : connectorInfoDTO.getOrgIdentifier())
```
Comment (abhinavsingh1196) [inline 126-instance/src/main/java/io/harness/service/instancesyncperpetualtask/instancesyncperpetualtaskhandler/k8s/K8SInstanceSyncPerpetualTaskHandler.java:184]: nit: StringUtils can clean this up
defaultIfEmpty(connectorInfoDTO.getOrgIdentifier(), EMPTY)

**Review by abhinavsingh1196 — APPROVED** [review]

File: 126-instance/src/main/java/io/harness/service/instancesyncperpetualtask/instancesyncperpetualtaskhandler/k8s/K8SInstanceSyncPerpetualTaskHandler.java (hunk 1 above)
Comment (namantalaycha10) [reply 126-instance/src/main/java/io/harness/service/instancesyncperpetualtask/instancesyncperpetualtaskhandler/k8s/K8SInstanceSyncPerpetualTaskHandler.java]: Done

**Comment by namantalaycha10** [comment]
trigger messagecheck

**Comment by namantalaycha10** [comment]
trigger uts

**Comment by namantalaycha10** [comment]
trigger uts

File: 950-delegate-tasks-beans/src/main/java/io/harness/perpetualtask/instancesync/DeploymentReleaseDetails.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2023 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+ */
+
+package io.harness.perpetualtask.instancesync;
```
Comment (abosii-harness) [inline 950-delegate-tasks-beans/src/main/java/io/harness/perpetualtask/instancesync/DeploymentReleaseDetails.java:8]: Can we move all these classes to their own package?

File: 950-delegate-tasks-beans/src/main/java/io/harness/perpetualtask/instancesync/DeploymentReleaseDetails.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import io.harness.annotations.dev.OwnedBy;
+
+import java.util.List;
+import lombok.Builder;
+import lombok.Data;
+@Data
+@Builder
+@OwnedBy(HarnessTeam.CDP)
+public class DeploymentReleaseDetails {
+  String taskInfoId;
```
Comment (abosii-harness) [inline 950-delegate-tasks-beans/src/main/java/io/harness/perpetualtask/instancesync/DeploymentReleaseDetails.java:20]: Either add default private or add missing access modifier private to all fields

**Review by GoelPratyush99 — APPROVED** [review]
lgtm

> Reviewed 2 of 11 files at r1, 20 of 25 files at r2, 2 of 3 files at r3, 1 of 3 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47465#-:-NVhgr6n4PfI67DERQbq:bb7k0qh)* status: 27 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team.  (waiting on @alebuh)

File: 950-delegate-tasks-beans/src/main/java/io/harness/perpetualtask/instancesync/CloudClusterConfig.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import io.harness.annotations.dev.HarnessTeam;
+import io.harness.annotations.dev.OwnedBy;
+
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+@OwnedBy(HarnessTeam.CDP)
+public class CloudClusterConfig {
```
Comment (abosii-harness) [inline 950-delegate-tasks-beans/src/main/java/io/harness/perpetualtask/instancesync/CloudClusterConfig.java:19]: Let's rename this to KubernetesCloudClusterConfig

**Review by abosii-harness — CHANGES_REQUESTED** [review]

**Comment by namantalaycha10** [comment]
trigger commonchecks

**Comment by namantalaycha10** [comment]
trigger uts

**Review by GoelPratyush99 — APPROVED** [review]
> Reviewed 15 of 15 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47465)* status: 28 file(s) to review. 1 change requests. 1 of 2 approvals obtained. Waiting on team(s): del-team.  (waiting on @alebuh)

**Review by abosii-harness — APPROVED** [review]

**Review by mdmilic — APPROVED** [review]
I don't think there are any files (that should be) code owned by del team. Approving to unblock.

**Comment by namantalaycha10** [comment]
trigger review

**Comment by namantalaycha10** [comment]
trigger pmd

**Comment by namantalaycha10** [comment]
trigger codebasehash

**Pull request merged**